src/_pytest/nodes.py

```diff
@@ -158,22 +150,24 @@ def __init__(
         #: Allow adding of extra keywords to use for matching.
         self.extra_keyword_matches: Set[str] = set()
 
-        if nodeid is not None:
-            assert "::()" not in nodeid
-            self._nodeid = nodeid
-        else:
-            if not self.parent:
-                raise TypeError("nodeid or parent must be provided")
-            self._nodeid = self.parent.nodeid
-            if self.name != "()":
-                self._nodeid += "::" + self.name
-
         # A place where plugins can store information on the node for their
         # own use. Currently only intended for internal plugins.
         self._store = Store()
 
+    @property
+    def fspath(self) -> py.path.local:
+        return py.path.local(self.fs_path)
```

Comment:
```suggestion
    def fspath(self) -> py.path.local:
        "Filesystem path where this node was collected from (can be None).
        
        Since pytest 6.2, prefer to use `fs_path` instead which returns a `pathlib.Path`
        instead of a `py.path.local`.
        """
        return py.path.local(self.fs_path)
```

Also for backward compat, should we add a (possibly deprecated) setter?

Reply:
i would take a risk there and ship without one first as its a fast addition